praw/models/reddit/draft.py

```diff
@@ -0,0 +1,331 @@
+                               `Unix Time`_.
+    ``original_content``       Whether the submission draft will be set as original
+                               content.
+    ``selftext``               The submission draft's selftext. ``None`` if a link
+                               submission draft.
+    ``spoiler``                Whether the submission will be marked as a spoiler.
+    ``subreddit``              Provides an instance of :class:`.Subreddit` or
+                               :class:`.UserSubreddit` (if set).
+    ``title``                  The title of the submission draft.
+    ``url``                    The URL the submission draft links to. Will b
+    ========================== ======================================================
+
+    .. _unix time: https://en.wikipedia.org/wiki/Unix_time
+
+    """
+
+    STR_FIELD = "id"
+
+    @classmethod
+    def _prepare_data(
+        cls,
+        flair_id: Optional[str] = None,
+        flair_text: Optional[str] = None,
+        is_public_link: Optional[bool] = None,
+        nsfw: Optional[bool] = None,
+        original_content: Optional[bool] = None,
+        selftext: Optional[str] = None,
+        send_replies: Optional[bool] = None,
+        spoiler: Optional[bool] = None,
+        subreddit: Optional[
+            Union["praw.models.Subreddit", "praw.models.UserSubreddit"]
+        ] = None,
+        title: Optional[str] = None,
+        url: Optional[str] = None,
+        **draft_kwargs,
+    ):
+        data = {
+            "body": selftext or url,
+            "flair_id": flair_id,
+            "flair_text": flair_text,
+            "is_public_link": is_public_link,
+            "kind": "markdown" if selftext is not None else "link",
+            "nsfw": nsfw,
+            "original_content": original_content,
+            "send_replies": send_replies,
+            "spoiler": spoiler,
+            "title": title,
+        }
+        if subreddit:
+            data.update(
+                {
+                    "subreddit": subreddit.fullname,
+                    "target": "profile"
+                    if subreddit.display_name.startswith("u_")
+                    else "subreddit",
+                }
+            )
+        data.update(draft_kwargs)
+        data = {key: value for key, value in data.items() if value is not None}
+        return data
+
+    def __init__(
+        self,
+        reddit: "praw.Reddit",
+        id: Optional[str] = None,  # pylint: disable=redefined-builtin
+        _data: Dict[str, Any] = None,
+    ):
+        """Construct an instance of the Draft object."""
+        if (id, _data).count(None) != 1:
+            raise TypeError("Exactly one of `id` or `_data` must be provided.")
+        fetched = False
+        if id:
+            self.id = id
+        else:
+            if len(_data) > 1:
+                if _data["kind"] == "markdown":
+                    _data["selftext"] = _data.pop("body")
```

Comment:
Is it weird that we're transforming these values? While it's more consistent with other Reddit models, I feel like we should probably go with whatever names Reddit uses (other than converting camelCase to snake_case where necessary).

Reply:
The issue with that is this endpoint uses `body` for both the selftext if a text post and url for a link post. Yes, my intention was to be consistent with the `submit` method. This also allows one to _submit_ a draft using the same arguments as the `Subreddit.submit` method. I'd have to double check, I think reddit requires `body` to be set when creating or modifying the draft.